examples/common/python/connectors/direct/tcs_listener/tcs_workorder_receipt_handler.py

```diff
@@ -35,311 +39,399 @@ def __init__(self, kv_helper):
+                )
+
+# ------------------------------------------------------------------------------------------------
+
+    def __validate_work_order_receipt_create_req(self, wo_receipt_req, wo_request):
+        """
+        Function to validate the work order receipt create request parameters
+        Parameters:
+            - wo_receipt_req is work order receipt request as dictionary
+            - wo_request is string containing jrpc work order request
+        Returns - tuple containing validation status(Boolean) and error message(string)
+        """
+        # Valid parameters list
+        valid_params = ["workOrderId", "workerServiceId", "workerId", "requesterId",
+                        "receiptCreateStatus", "workOrderRequestHash", "requesterGeneratedNonce",
+                        "requesterSignature", "signatureRules","receiptVerificationKey"]
+        for key in wo_receipt_req["params"]:
+            if key not in valid_params:
+                return False, "Missing parameter " + key + " in the request"
+            else:
+                if key in ["workOrderId", "workerServiceId", "workerId",
+                           "requesterId", "requesterGeneratedNonce"]:
+                    if not is_valid_hex_str(wo_receipt_req["params"][key]):
+                        return False, "invalid data parameter for " + key
+                elif key in ["workOrderRequestHash", "requesterSignature"]:
+                    try:
+                        base64.b64decode(wo_receipt_req["params"][key])
+                    except Exception as e:
+                        return False, \
+                            "Invalid data format for " + key
+        receipt_type = wo_receipt_req["params"]["receiptCreateStatus"]
+        try:
+            receipt_enum_type = ReceiptCreateStatus(receipt_type)
+        except:
+            return False, "Invalid receipt status type"
+
+        # Validate signing rules
+        signing_rules = wo_receipt_req["params"]["signatureRules"]
+        rules = signing_rules.split("/")
+        if len(rules) == 2 and rules[0] != self.HASHING_ALGORITHM and \
+            rules[1] != self.SIGNING_ALGORITHM:
+            return False, "Unsupported the signing rules"
+
+        signature_obj = signature.ClientSignature()
+        # Verify work order request is calculated properly or not.
+        wo_req_hash = signature_obj.calculate_request_hash(json.loads(wo_request))
+        if wo_req_hash != wo_receipt_req["params"]["workOrderRequestHash"]:
+            return False, "Work order request hash does not match"
+        # Verify requester signature with signing key in the request
+        status = signature_obj.verify_create_receipt_signature(wo_receipt_req)
+        if status != SignatureStatus.PASSED:
+            return False, "Receipt create requester signature does not match"
+        # If all parameters are verified in the request
+        return True, ""
```

Comment:
Can we add message(say success) after True value?

Reply:
we are not using the second argument if it success. In case error second argument will have different error messages.

Reply:
Ok